datamade/just-spaces
# Preview functionality

## Overview

Adds ability to preview unpublished surveys. https://github.com/datamade/django-fobi/pull/3 should be reviewed and merged before this PR.

Success messages will need some work in order to be customizable and indicate when forms are in preview mode. I've opened an issue for that: #108 

## To do

- [x] Once the `django-fobi` PR linked above is merged: write a test!

## Testing Instructions

 * If you don't already have it, create the database: `pg_restore -C -j4 --no-owner just-spaces.dump | psql`
* Migrate: `python manage.py migrate`
* Login as testuser (password in the DataMade LastPass under "Just Spaces Staging")
* Create a new survey
* Add at least one form element to it
* Go back to the list view and hit the "Preview button" 
* Does it say "Preview [Survey Name] at the top of that detail page?
* Submit your survey
* Check the collected data list view for your survey and make sure submitted data _isn't_ there

Connects #89
